### DISCUSSION TITLE
Expansion filters depending on focused window

### DISCUSSION THREAD

**Discussion by tomsterBG** [psoukie/zipchord#167] [Ideas]
Would it be possible to blacklist/whitelist what chords/shorthands expand to depending on currently focused window? 
I'm new to ahk so I don't yet know what can be done with it. 
For example: 
- Let some things work on everything, except when you open a code editor. Let's say space and capitalization after a dot.
- The fn chord would expand to function in code, but to fine in text. The filter would (respectively) check if the focused window is or isn't for example VSCode or Notepad++.

If that's not possible then maybe a dictionary that takes into account context to figure out what type of text you're in. That's definitely doable as you just need to copy text and use that as an input to the filter.
For example: 
- If previous text contains local or function it's most likely code.
- If previous text contains 's, I, why or hi it's most likely text.

**Comment by psoukie** [comment]
Yes, it would definitely be possible to modify the active dictionary (or dictionaries) based on what window is active. I'll add it to the roadmap.
